src/main/java/uk/gov/hmcts/reform/security/keyvault/KeyVaultKeyStore.java

```diff
@@ -42,9 +60,26 @@ public Key engineGetKey(String alias, char[] password) {
             }
         } else {
             // Try looking up a secret based on this alias
-            // TODO may need to support this functionality in the future
-            throw new ProviderException("Secret-based keys are not implemented");
+            SecretBundle bundle = vaultService.getSecretByAlias(alias);
+            if (bundle != null) {
+                KeyStore.SecretKeyEntry entry = new KeyStore
+                    .SecretKeyEntry(new SecretKeySpec(bundle.value().getBytes(), "RAW"));
```

Comment:
is `RAW` a valid value for the secret key algorithm?

Reply:
@nikola-naydenov-hmcts confirmed it's valid and being used by ForgeRock AM when creating secret keys in JKS keystores.